File: src/design_patterns/creational/builder/php/pizza/PizzaBuilder.php
<?php

abstract class PizzaBuilder {

    /**
     * @var Pizza
     */
    protected $pizza;

    /**
     * @return mixed
     */
    public function getPizza() {
        return $this->pizza;
    }

    /**
     * Set a new Pizza object
     */
    public function createNewPizzaProduct() {
        $this->pizza = new Pizza();
    }

    abstract public function buildDough();

    abstract public function buildSauce();

    abstract public function buildTopping();
}

File: src/design_patterns/creational/builder/java/Pizza/PizzaBuilder.java
/* "Abstract Builder" */
abstract class PizzaBuilder {
	protected Pizza pizza;

	public Pizza getPizza() { return pizza; }
	public void createNewPizzaProduct() { pizza = new Pizza(); }

	public abstract void buildDough();
	public abstract void buildSauce();
	public abstract void buildTopping();
}

